File: rldp-diceapp-ui/src/app/core/models/dice-response-group.ts
import { DiceSumDistribution } from './dice-sum-distribution';
import { DiceResponse } from './dice-response';
import { DistinctRequest } from './distinct-request';

export class DiceResponseGroup {

    totalRolls: number;
    totalSimulations: number;
    diceSumDistributionList: Array<DiceSumDistribution> = [];
    diceResponseList: Array<DiceResponse> = [];
    request: DistinctRequest;

    constructor (param?: any) {
        this.totalRolls = param.totalRolls;
        this.totalSimulations = param.totalSimulations;
        this.request = new DistinctRequest(param.request);
        param.diceSumDistributionList.forEach(element => {
            this.diceSumDistributionList.push(new DiceSumDistribution(element));
        });
        param.diceResponseList.forEach(element => {
            this.diceResponseList.push(new DiceResponse(element));
        });
    }
}

File: rldp-diceapp-service/src/main/java/com/rldp/diceapp/model/DiceResponseGroup.java
package com.rldp.diceapp.model;

import java.util.ArrayList;
import java.util.List;

public class DiceResponseGroup {

	private int totalRolls;
	private int totalSimulations;
	private List<DiceSumDistribution> diceSumDistributionList;

	private final List<DiceResponse> diceResponseList = new ArrayList<>();

	private final DistinctRequest request;

	public DiceResponseGroup(DistinctRequest request) {
		this.request = request;
	}

	public List<DiceResponse> getDiceResponseList() {
		return diceResponseList;
	}

	public DistinctRequest getRequest() {
		return request;
	}
	
	public int getTotalRolls() {
		return totalRolls;
	}
	
	public int getTotalSimulations() {
		return totalSimulations;
	}

	public List<DiceSumDistribution> getDiceSumDistributionList() {
		return diceSumDistributionList;
	}

	public void setDiceSumDistributionList(List<DiceSumDistribution> diceSumDistributionList) {
		this.diceSumDistributionList = diceSumDistributionList;
	}

	public void add(DiceResponse response) {
		diceResponseList.add(response);
		totalRolls += response.getDiceRollList().size();
		totalSimulations += 1;
	}
}
